isthmus/src/main/java/io/substrait/isthmus/SubstraitRelNodeConverter.java

```diff
@@ -89,22 +109,128 @@ public RelNode visit(NamedScan namedScan) throws RuntimeException {
+    var condition =
+        join.getCondition()
+            .map(c -> c.accept(expressionRexConverter))
+            .orElse(relBuilder.literal(true));
+    var joinType =
+        switch (join.getJoinType()) {
+          case INNER -> JoinRelType.INNER;
+          case LEFT -> JoinRelType.LEFT;
+          case RIGHT -> JoinRelType.RIGHT;
+          case OUTER -> JoinRelType.FULL;
+          case SEMI -> JoinRelType.SEMI;
+          case ANTI -> JoinRelType.ANTI;
+          case UNKNOWN -> throw new UnsupportedOperationException(
+              "Unknown join type is not supported");
+        };
+    return relBuilder.push(left).push(right).join(joinType, condition).build();
+  }
+
+  @Override
+  public RelNode visit(Aggregate aggregate) throws RuntimeException {
+    RelNode child = aggregate.getInput().accept(this);
+    var groupExprLists =
+        aggregate.getGroupings().stream()
             .map(
-                expr ->
-                    expr.accept(
-                        new ExpressionRexConverter(
-                            relOptCluster.getTypeFactory(), scalarFunctionConverter)))
+                gr ->
+                    gr.getExpressions().stream()
+                        .map(expr -> expr.accept(expressionRexConverter))
+                        .toList())
             .toList();
+    List<RexNode> groupExprs =
+        groupExprLists.stream().flatMap(Collection::stream).collect(Collectors.toList());
+    RelBuilder.GroupKey groupKey = relBuilder.groupKey(groupExprs, groupExprLists);
 
-    return relBuilder.push(child).project(rexList).build();
+    List<AggregateCall> aggregateCalls =
+        aggregate.getMeasures().stream().map(m -> fromMeasure(m)).toList();
+    return relBuilder.push(child).aggregate(groupKey, aggregateCalls).build();
+  }
+
+  private AggregateCall fromMeasure(Aggregate.Measure measure) {
+    var arguments =
+        measure.getFunction().arguments().stream()
+            .map(expr -> expr.accept(expressionRexConverter))
+            .toList();
+    var operator =
+        aggregateFunctionConverter.getSqlOperatorFromSubstraitFunc(
+            measure.getFunction().declaration().key(), measure.getFunction().outputType());
+    if (!operator.isPresent()) {
+      throw new UnsupportedOperationException(
+          String.format(
+              "Unable to find binding for call %s", measure.getFunction().declaration().name()));
+    }
+    List<Integer> argIndex = new ArrayList<>();
+    for (RexNode arg : arguments) {
+      if (!(arg instanceof RexInputRef)) {
+        throw new UnsupportedOperationException(
+            String.format(
+                "aggregation function only support column as argument. % is not support", arg));
+      }
+      argIndex.add(((RexInputRef) arg).getIndex());
+    }
+
+    boolean distinct =
+        measure.getFunction().invocation()
+            == AggregateFunction.AggregationInvocation.AGGREGATION_INVOCATION_DISTINCT;
```

Comment:
Seems like we should put a check here to ensure invocation is not some unexpected type.

Reply:
If invocation is other type,  `distinct` is `false`.   For what reason, we need put such check?

Reply:
The problem is someone may introduce another branch of the enum in the future. If they do, this path will blindly ignore it. (Basically, the default case of switch).

Reply:
Changed to use switch statement with default cases.